src/extension-runners/firefox-desktop.js

```diff
@@ -0,0 +1,229 @@
+/* @flow */
+
+// Import flow types from npm dependencies.
+import type FirefoxProfile from 'firefox-profile';
+
+import {
+  MultipleExtensionsReloadError,
+  RemoteTempInstallNotSupported,
+  WebExtError,
+} from '../errors';
+import * as defaultFirefoxApp from '../firefox';
+import {
+  connectWithMaxRetries as defaultFirefoxConnector,
+} from '../firefox/remote';
+import {createLogger} from '../util/logger';
+// Import flow types from project files.
+import type {
+  FirefoxRDPResponseAddon,
+  RemoteFirefox,
+} from '../firefox/remote';
+import type {
+  ExtensionRunnerParams,
+} from './base';
+import type {
+  FirefoxPreferences,
+} from '../firefox/preferences';
+import type {
+  FirefoxInfo, // eslint-disable-line import/named
+} from '../firefox/index';
+
+export type FirefoxDesktopExtensionRunnerParams = ExtensionRunnerParams & {
```

Comment:
@kumar303 I just tried and it works (the tests will require a bunch of additional $FLOW_IGNORE on some of the sinon spy, but, on the bright side, flow is able to catch a lot more potential issue in the code).

We have to update flow-bin to use the feature from the above link, which currently raises a warning during npm install because of the peer dependencies currently defined in grunt-flowbin:

https://github.com/kumar303/grunt-flowbin/blob/master/package.json#L39

we should update the peer dependency to `"flow-bin": ">=0.47.0"`, while we are still working on this review, so that when this PR is ready we can update the grunt-flowbin version in the web-ext devDependencies and remove the warning.